vars/getTestResults.groovy

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import hudson.tasks.test.AbstractTestResultAction
+import groovy.json.JsonOutput
+import java.text.SimpleDateFormat
+import java.util.Date
+
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+    def finalJsonDoc = ""
+    currentBuild.description = "runner: Ubuntu2004 M58xlarge<br><a href=\"https://github.com/opensearch-project/OpenSearch/pull/12503\">PR #12503</a>: removing unused fetch sub phase processor initialization during fetch"
+    def buildNumber = currentBuild.number
+    def buildDescription = currentBuild.description
+    def buildDuration = currentBuild.duration
+    def buildResult = currentBuild.result
+    def buildStartTime = currentBuild.startTimeInMillis
+    def prString = getPrAndDescription(buildDescription)[0]
+    def remainingText = getPrAndDescription(buildDescription)[1]
+    def currentDate = new Date()
+    def formattedDate = new SimpleDateFormat("MM-yyyy").format(currentDate)
+
+    println("The formatted date is ${formattedDate}")
+    def indexName = "gradle-check-${formattedDate}"
+
+
+    def test_docs = getFailedTestRecords(buildNumber, prString, remainingText, buildResult, buildDuration, buildStartTime)
+
+    if (test_docs) {
+        for (doc in test_docs) {
+            def jsonDoc = JsonOutput.toJson(doc)
+            finalJsonDoc += "{\"index\": {\"_index\": \"${indexName}\"}}\n" + "${jsonDoc}\n"
+        }
+        writeFile file: "failed-test-records.json", text: finalJsonDoc
+
+        def fileContents = readFile(file: "failed-test-records.json").trim()
+        println("File Content is:\n${fileContents}")
+        indexFailedTestData()
+    }
+}
+
+List<String> getPrAndDescription(String buildDescription){
+    List<String> pr_details = []
+    // Fetch the PR number and description. In case the gradle-check is run as part of post-merge
+    // action then PR # will be "other" and description will contain string that a commit was pushed
+    def pattern = /<a.*?>(.*?)<\/a>:\s*(.*)$/
+    def matcher = (buildDescription =~ pattern)
+    if (matcher.find()) {
+        def prString = matcher[0][1]
+        def remainingText = matcher[0][2]
+        pr_details.add(prString.toString())
+        pr_details.add(remainingText.toString())
+        return pr_details
+    }
+    return ["", ""]
```

Comment:
So if the build description of the gradle check changes, this also needs to be updated according from now on?

Reply:
Maybe you can update gradle check workflow to include some hidden parameters instead of relying on description?

Reply:
Looking the current code the job description is pretty uniform and unlikely to change. See https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/gradle-check.yml#L30-L50. 

Not really sure if adding the same as hidden parameter would benefit much and I also do not want to modify `gradle-check` job. 
I think I can use the existing [payload data](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/gradle-check.yml#L49-L50) as parameters to post run script.